### DISCUSSION TITLE
GenMon working with Cummins Gennies

### DISCUSSION THREAD

**Discussion by EvanET13** [jgyates/genmon#1115] [General]
Hi all,

I was curious to see if anyone has gotten GenMon to work with any older Cummins Generators. 
I have a Cummins Generator that uses their $600 PowerCommand Cloud Link aka PC500CLN (PN: A059Y211). But also have similar Cummins generators elsewhere that I would like to monitor for a fraction of the price and that I can do, instead of having to pay another $6000 to have a tech install the device (PC500). 

My generator info is as follows:
- Brand: Cummins Power Generation
- Model: GGMC
- Control Model: PCC1301
- Frequency: 60Hz
![image](https://github.com/jgyates/genmon/assets/60356533/48dbac69-9366-49fc-9411-ba126ef13d35)

It is accompanied by an ATS:
- Model: OHPCB
- Control Model: OTPC
![image](https://github.com/jgyates/genmon/assets/60356533/9b655fd8-f2da-4d96-8fea-5f46f510d964)

I understand I may have to use a custom HAT with relays, but I'm not sure how to physically set it up and want to get as much information to monitor as possible.

**Comment by jgyates** [comment]
Hi @EvanET13 

To date, no support for Cummins controllers have been added to genmon. Also, genmon only supports controllers that support modbus, either modbus TCP or modbus RTU (serial).

To add a new controller the following steps are typically the path that is taken

1) Acquire modbus register definitions. There are some Cummins modles that have modbus register definitions on the web, but I don't claim to know anything about Cummins models that support modbus.
2) Dump the contents of the modbus registers with the (modbusdump.py)[https://github.com/jgyates/genmon/wiki/1----Software-Overview#otherappsmodbusdumppy] program in the project repository.
3) Once I have number 1 and 2 I can typically add support for the controller via the (custom controller interface)[https://github.com/jgyates/genmon/wiki/Appendix-N-Genmon-Supporting-Other-Controller-Types].
